### PR TITLE
feat: add reset method for useSetState

### LINKED ISSUE #1776: [RFC] useResetState
# 功能
跟 `React.useState` 基本一致，但是返回一个可以重置state初始值的方法

```
const [state, setState, resetState] = useResetState(initialState)

resetState() // state => initialState
```

同理，`useSetState` 也可以加此功能

解决痛点：`resetState` 在业务场景中很常见，可是开发者不得不写一份初始值，在代码维护性上也不太好

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

close #1776 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | add reset method for useSetState |
| 🇨🇳 Chinese | 给 useSetState 新增 reset 方法 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
从master拉的新分支……不知道为啥会有这么多次commit，离谱……

File: packages/hooks/src/useSetState/index.zh-CN.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ nav:
 
 # useSetState
 
-管理 object 类型 state 的 Hooks，用法与 class 组件的 `this.setState` 基本一致。
+管理 object 类型 state 的 Hooks，用法与 class 组件的 `this.setState` 基本一致。与 `this.setState` 相比多了一个 `reset` 方法，用来将状态重置为初始值。
```
Comment (yaogengzhu) [inline packages/hooks/src/useSetState/index.zh-CN.md]: 与 `this.setState` 相比多了一个 `resetState` 方法，用来将状态重置为初始值。

**Comment by crazylxr** [comment]
看下 issue，暂时就 close 了

**Comment by li-jia-nan** [comment]
> 看下 issue，暂时就 close 了

OK，看到啦

**Pull request closed without merging**